### PR TITLE
GG-33019 Introduce ProjectableFilterableTableScan

### PR DESCRIPTION
**Pull request by tmagomedov**
For native storage integration.

### CODE REVIEW

File: modules/sql/pom.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        <version>1</version>
+        <relativePath>../../parent/pom.xml</relativePath>
+    </parent>
+
+    <artifactId>ignite-sql</artifactId>
+    <version>3.0.0-SNAPSHOT</version>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.ignite</groupId>
```
Comment (vveider) [inline modules/sql/pom.xml:37]: Internal dependencies should also be declared in parent pom's dependencyManagement section with version and here without.

File: parent/pom.xml [hunk 2]
```diff
@@ -74,6 +74,7 @@
         <typesafe.version>1.4.1</typesafe.version>
         <hamcrest.version>2.2</hamcrest.version>
         <scalecube.version>2.6.6</scalecube.version>
+        <calcite.version>1.26.0</calcite.version>
```
Comment (vveider) [inline parent/pom.xml:77]: Version properties should be ABC sorted.

**Review by vveider — CHANGES_REQUESTED** [review]
It seems that after this commit, we will have to do IGNITE => CE sync with cherry-pick instead of FF?

File: modules/sql/src/main/java/org/apache/ignite/internal/sql/rel/IgniteTableScan.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.List;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.util.ImmutableBitSet;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ *
+ */
+public class IgniteTableScan extends ProjectableFilterableTableScan {
+    public IgniteTableScan(
```
Comment (AMashenkov) [inline modules/sql/src/main/java/org/apache/ignite/internal/sql/rel/IgniteTableScan.java:29]: Javadoc

File: modules/sql/src/main/java/org/apache/ignite/internal/sql/rel/ProjectableFilterableTableScan.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        @Nullable List<RexNode> proj,
+        @Nullable RexNode cond,
+        @Nullable ImmutableBitSet reqColumns
+    ) {
+        projects = proj;
+        condition = cond;
+        requiredColumns = reqColumns;
+    }
+
+    /** @return Projections. */
```
Comment (AMashenkov) [inline modules/sql/src/main/java/org/apache/ignite/internal/sql/rel/ProjectableFilterableTableScan.java:47]:
```suggestion
    /** 
    * @return Projections. 
    */
```

File: modules/sql/src/main/java/org/apache/ignite/internal/sql/rel/ProjectableFilterableTableScan.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    public List<RexNode> projects() {
+        return projects;
+    }
+
+    /** @return Rex condition. */
```
Comment (AMashenkov) [inline modules/sql/src/main/java/org/apache/ignite/internal/sql/rel/ProjectableFilterableTableScan.java:52]:
```suggestion
    /** 
    * @return Rex condition. 
    */
```

File: modules/sql/src/main/java/org/apache/ignite/internal/sql/rel/ProjectableFilterableTableScan.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    public RexNode condition() {
+        return condition;
+    }
+
+    /** @return Participating columns. */
```
Comment (AMashenkov) [inline modules/sql/src/main/java/org/apache/ignite/internal/sql/rel/ProjectableFilterableTableScan.java:57]:
```suggestion
    /** 
    * @return Participating columns. 
    */
```

File: modules/sql/src/main/java/org/apache/ignite/internal/sql/rel/ProjectableFilterableTableScan.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    public ImmutableBitSet requiredColumns() {
+        return requiredColumns;
+    }
+
+    /** */
```
Comment (AMashenkov) [inline modules/sql/src/main/java/org/apache/ignite/internal/sql/rel/ProjectableFilterableTableScan.java:62]: Javadoc.

**Comment by tmagomedov** [comment]
Looks like it is better to leave this repo code same as ignite-3 for now, closing merge request.

**Pull request closed without merging**